android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/login/LoginActivityTest.java

```diff
@@ -170,10 +164,29 @@ public void testValidInput() {
 
         onView(withId(R.id.login_password_layer))
                 .check(matches(not(hasTextInputLayoutErrorText(appContext.getString(R.string.error_message_password_field_empty)))));
+    }
+
+    @Test
+    public void testPasswordVisibilityToggle() {
+        onView(withId(R.id.login_password))
+                .perform(typeText("testPassword"), closeSoftKeyboard());
 
+        // tests that the password is not readable
+        activityScenarioRule.getScenario().onActivity(activity -> {
+            EditText passwordFieldText = activity.findViewById(R.id.login_password);
+            assertNotEquals("testPassword", passwordFieldText.getLayout().getText().toString());
+        });
 
-    }
+        // click on the visibility toggle
+        onView(withContentDescription("password_toggle"))
+                .perform(click());
 
+        // tests that the password is readable
+        activityScenarioRule.getScenario().onActivity(activity -> {
+            EditText passwordFieldText = activity.findViewById(R.id.login_password);
+            assertEquals("testPassword", passwordFieldText.getLayout().getText().toString());
+        });
```

Comment:
Here you could use a variable (e.g. String password = "testPassword") but I think its not necessary